### DISCUSSION TITLE
Why do we need shiny-binding-wrapper?

### DISCUSSION THREAD

**Discussion by jonekeat** [Appsilon/shiny.react#81] [Q&A] [answered]
For instance, I noticed `shiny.react` now checks if the className of elements contains the following patterns
https://github.com/Appsilon/shiny.react/blob/43679fa556d14dfc9fb7f0cdeefa53375801e3b0/js/src/react/shinyBindings.jsx#L5-L16

if yes, then the package will render the element with a div wrapper `shiny-binding-wrapper` as in 
https://github.com/Appsilon/shiny.react/blob/43679fa556d14dfc9fb7f0cdeefa53375801e3b0/js/src/react/shinyBindings.jsx#L22-L37

this may cause issues to layout / action if user incorporated their CSS & JS code, and defined custom className, here is an example:

### Use className `action-button`
This looks weird, when we used class 'action-button'

![image](https://github.com/Appsilon/shiny.react/assets/51594799/c70bf268-eebc-45be-b645-4d5d5e0708ea)

```r
library(shiny)
library(shiny.fluent)

shinyApp(
  ui = fluentPage(
    div(
      class = "example-card action-button",
      style = "margin-top: 10px; margin-bottom: 10px; width: max(450px, 35%);", # width: fit-content;
      div(
        class = "example-card-header action-button",
        style = "border-bottom: 1px solid rgb(161, 159, 157); overflow: hidden;",
        Stack(
          className = "example-card-header-stack",
          horizontal = TRUE,
          horizontalAlign = "left",
          tokens = list(childrenGap = 10L),
          list(
            h3(
              class = "example-card-title action-button",
              style = "font-size: 20px; font-weight: 600; color: #605e5c; display: inline-flex; flex-grow: 1; flex-shrink: 1; margin-bottom: 10px;",
              "Action Button"
            ),
            CommandButton.shinyInput(
              inputId = "code_action_btn",
              className = "example-card-code-button action-button",
              styles = list(
                root = list(
                  alignSelf = "flex-end",
                  borderWidth = "1px 1px 0px"
                )
              ),
              iconProps = list("iconName" = "Embed"),
              disabled = FALSE,
              text = "Show code"
            )
          )
        )
      )
    )
  ),
  server = function(input, output) {}
)
```

### Use className `action-btn`
This looks normal, when we change class to 'action-btn'

![image](https://github.com/Appsilon/shiny.react/assets/51594799/f20d328f-b5ec-43bc-9fe0-92507a5d393b)

```r
library(shiny)
library(shiny.fluent)

shinyApp(
  ui = fluentPage(
    div(
      class = "example-card action-btn",
      style = "margin-top: 10px; margin-bottom: 10px; width: max(450px, 35%);", # width: fit-content;
      div(
        class = "example-card-header action-btn",
        style = "border-bottom: 1px solid rgb(161, 159, 157); overflow: hidden;",
        Stack(
          className = "example-card-header-stack",
          horizontal = TRUE,
          horizontalAlign = "left",
          tokens = list(childrenGap = 10L),
          list(
            h3(
              class = "example-card-title action-btn",
              style = "font-size: 20px; font-weight: 600; color: #605e5c; display: inline-flex; flex-grow: 1; flex-shrink: 1; margin-bottom: 10px;",
              "Action Button"
            ),
            CommandButton.shinyInput(
              inputId = "code_action_btn",
              className = "example-card-code-button action-btn",
              styles = list(
                root = list(
                  alignSelf = "flex-end",
                  borderWidth = "1px 1px 0px"
                )
              ),
              iconProps = list("iconName" = "Embed"),
              disabled = FALSE,
              text = "Show code"
            )
          )
        )
      )
    )
  ),
  server = function(input, output) {}
)
```

I just dont understand why we need this logic here, please let me know if i misunderstood anything @kamilzyla @jakubnowicki 
I believe this is related to https://github.com/Appsilon/shiny.react/issues/52

**Comment by jakubsob** [comment, marked as answer]
Hi @jonekeat, yes, this is related to #52.

This wrapper was made to make it simpler to render Shiny bound nodes. A rendering mechanism of those nodes must be changed for us to be able to remove that wrapper. 

There is work in progress on that issue in https://github.com/Appsilon/shiny.react/pull/83, but it requires thorough testing before it's merged to ensure that all dependent packages work as expected with that change.

**Comment by jonekeat** [comment]
thanks @jakubsob for your clarification
